Fix flaky test_reload_while_loading

tuanpach opened the pull request:
<!---
A technical comment, you are free to remove or leave it as it is when PR is created
The following categories are used in the next scripts, update them accordingly
utils/changelog/changelog.py
tests/ci/cancel_and_rerun_workflow_lambda/app.py
-->
### Changelog category (leave one):
- CI Fix or Improvement (changelog entry is not required)


### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix test_reload_while_loading, increase timeout from 1 to 10s, ensure that the instance receives the query triggering the dict loading

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

<!---
Directly edit documentation source files in the "docs" folder with the same pull-request as code changes

or

Add a user-readable short description of the changes that should be added to docs.clickhouse.com below.

At a minimum, the following information should be added (but add more as needed).
- Motivation: Why is this function, table engine, etc. useful to ClickHouse users?

- Parameters: If the feature being added takes arguments, options or is influenced by settings, please list them below with a brief explanation.

- Example use: A query or command.
-->

tuanpach committed:
Fix test_reload_while_loading, increase timeout from 1 to 2s, ensure that the instance receive the query triggering the dict loading

tuanpach committed:
Fix flaky test_dictionaries_update_and_reload

azat commented on tests/integration/test_dictionaries_update_and_reload/test.py:
```diff
@@ -82,64 +84,93 @@ def get_loading_duration(dictionary_name):
+        query(
+            "SELECT dictGetInt32('slow', 'a', toUInt64(5))",
+            query_id=query_id,
+            timeout=2,
```
2 seconds still looks too flaky (I am actually surprised that it does not fail with sanitizers), let's increase it to 10 at least, the slow dictionary uses `sleep 10`

Or instead we can use `max_execution_time` instead

azat replied:
OK, I do not wonder anymore, it works as expected, there is no need to increase this timeout, the failure was for a different reason

tuanpach replied:
I increased it to 10s.

tuanpach committed:
Increase timeout from 2s to 10s

azat commented:
>[Integration tests (release, 3/4)](https://s3.amazonaws.com/clickhouse-test-reports/json.html?PR=78511&sha=daf2e74008e91f7531b502f3d77aefb380a887e3&name_0=PR&name_1=Integration%20tests%20%28release%2C%203%2F4%29) — fail: 1, passed: 886

- https://github.com/ClickHouse/ClickHouse/issues/71049

tuanpach committed:
Remove checking duration and prev_duration

tuanpach commented:
I removed the check `duration < prev_duration` after `SYSTEM RELOAD DICTIONARY 'slow'`.

tuanpach commented:
- https://github.com/ClickHouse/ClickHouse/issues/78103
- https://github.com/ClickHouse/ClickHouse/issues/78542
- https://github.com/ClickHouse/ClickHouse/issues/78569

azat commented:
>I removed the check duration < prev_duration after SYSTEM RELOAD DICTIONARY 'slow'.

This test is odd BTW, yes, this [failure](https://pastila.nl/?0108383a/6e523c0012bb8c3181c5d28b2db21472#IIxVo8j09McA3bXlmKvnEA==) pops up due to the timeout had been increased to 10 seconds (likely), but:
- it still should be bigger, since we `sleep(1)` in the test
- why the test even cares about this? (cc @vitlibar )
- why it does not check the duration load after it has been replaced with `sleep 0`?

But for now I think it is OK

tuanpach commented:
I think the test wants to verify that the duration keeps increasing if the loading is not restarted. However, after restarting, the duration is reset, and we should not check it.

After we increased the timeout and added the `wait_for_log_line`, the assert sometimes failed. That's why I removed it.

tuanpach commented:
@azat Should we merge this PR?

azat commented:
I'm OK with it (we can polish the test later)
